### PR TITLE
Test/feat

### PR DESCRIPTION
**Pull request by rctz**

### CODE REVIEW

**Review by rctz — COMMENTED** [review]
It looks like the changes introduced contain a large number of duplicate lines. This duplication seems unintentional and could introduce issues, as redundant code may lead to unnecessary computations and might confuse future maintainers.

Here's a summary of problematic duplications:

1. **`currect_dir.push(".env");`**:
    - This line is repeated three times. It appears unnecessary to push the same `.env` multiple times onto the `currect_dir` path buffer.

2. **Environment variables retrieval (`env::var`)**:
    - Multiple sections are retrieving the same environment variables (`TABLE_NAME`, `CSV_OUTPUT_PREFIX`) redundantly. This duplicates logic unnecessarily and leads to clutter in the code.

3. **`output_path`** duplicated block:
    - The `let output_path` definition is repeated, and again the same environment variables (`CSV_OUTPUT_PREFIX` and `TABLE_NAME`) are redundantly fetched before it.

### Suggested Fixes:
Below is a cleaner and corrected version of the code where duplication is removed:

```rust
async fn main() -> Result<(), Box<dyn Error>> {
    let mut currect_dir = env::current_dir()
        .map_err(|_| "Failed to get current directory")?
        .to_path_buf();

    currect_dir.push(".env"); // Push `.env` only once.

    let env_path = Path::new(&currect_dir);
    dotenvy::from_path(env_path).ok();

    // Fetch environment variables only once and reuse them.
    let table_name_list_str =
        env::var("TABLE_NAME").map_err(|_| "Missing environment variable: TABLE_NAME")?;
    let csv_output_prefix =
        env::var("CSV_OUTPUT_PREFIX").map_err(|_| "Missing environment variable: CSV_OUTPUT_PREFIX")?;
    let output_path =
        env::var("OUTPUT_PATH").map_err(|_| "Missing environment variable: OUTPUT_PATH")?;

    let full_database_url = format!(
        "{}://{}:{}@{}:{}/{}",
        ...
    );

    // The rest of your code logic follows here.
}
```

### Key Changes:
1. **Removed repeated `.env` pushes**:
   - `.env` should only be appended once to the `currect_dir`.

2. **Consolidated environment variable retrieval**:
   - Environment variables are fetched once (`TABLE_NAME`, `CSV_OUTPUT_PREFIX`, and `OUTPUT_PATH`) and reused as needed. This keeps the code clean and removes redundancy.

3. **Improved readability**:
   - Consolidated logic makes the flow simpler and easier to follow.

Ensure you test the updated code to verify that the cleanup hasn't introduced breaking changes.

**Review by rctz — COMMENTED** [review]
It seems like there's a lot of repetitive code introduced in the changes. Specifically, the same variables (`TABLE_NAME`, `CSV_OUTPUT_PREFIX`) and their respective `env::var` calls are being repeatedly declared and initialized multiple times. Additionally, there’s some redundant `currect_dir.push(".env")` calls in the `main` function, which also seem unnecessary.

Here’s an analysis and some recommendations:

---

### Issues Identified:
1. **Duplicate `.env` push logic:**
   ```rust
   currect_dir.push(".env");
   currect_dir.push(".env");
   currect_dir.push(".env");
   ```
   This is redundant and serves no additional purpose.

2. **Repeated `env::var` calls for the same variables:**
   ```rust
   let table_name_list_str = env::var("TABLE_NAME")
       .map_err(|_| "Missing environment variable: TABLE_NAME")?;
   let csv_output_prefix = env::var("CSV_OUTPUT_PREFIX")
       .map_err(|_| "Missing environment variable: CSV_OUTPUT_PREFIX")?;
   ```
   These lines are repeated across multiple areas unnecessarily, which can be consolidated.

3. **Unused variables (`let output_path`):**
   In the final section of the diff, `let output_path` is partially introduced and seems incomplete/unused.

4. **Readability & Maintainability:**
   The repeated code impacts the code's readability and maintainability. Repeatedly retrieving the same environment variables in multiple places is inefficient and error-prone.

---

### Suggested Fix:

Here’s a cleaned-up version of your code that removes redundancies and organizes the logic better:

```rust
use std::env;
use std::error::Error;
use std::path::Path;

async fn main() -> Result<(), Box<dyn Error>> {
    let mut current_dir = env::current_dir()?;
    current_dir.push(".env");

    let env_path = Path::new(&current_dir);
    dotenvy::from_path(env_path).ok(); // Load environment variables

    // Retrieve and validate all required environment variables upfront
    let table_name_list_str =
        env::var("TABLE_NAME").map_err(|_| "Missing environment variable: TABLE_NAME")?;
    let csv_output_prefix =
        env::var("CSV_OUTPUT_PREFIX").map_err(|_| "Missing environment variable: CSV_OUTPUT_PREFIX")?;
    let output_path =
        env::var("OUTPUT_PATH").map_err(|_| "Missing environment variable: OUTPUT_PATH")?;

    // (Example of where you might want to use these variables)
    let full_database_url = format!(
        "{}://{}:{}@{}:{}/{}",
        env::var("DB_PROTOCOL").unwrap_or_default(),
        env::var("DB_USER").unwrap_or_default(),
        env::var("DB_PASSWORD").unwrap_or_default(),
        env::var("DB_HOST").unwrap_or_default(),
        env::var("DB_PORT").unwrap_or_default(),
        table_name_list_str    // Example usage of the retrieved `TABLE_NAME`
    );

    println!("CSV Output Prefix: {}, Output Path: {}", csv_output_prefix, output_path);
    // You can now use `table_name_list_str`, `csv_output_prefix`, and `output_path` in your logic

    Ok(())
}
```

---

### Key Improvements:

1. **Avoiding Redundancy:**
   - Added a single `env::var` call for each required environment variable (`TABLE_NAME`, `CSV_OUTPUT_PREFIX`, `OUTPUT_PATH`) instead of repeating them.
   - Consolidated the `.env` path logic: instead of repeatedly appending `.env` to `current_dir`, it’s appended just once.

2. **Improved Readability:**
   - The cleaned-up version makes it clear what environment variables are being used and ensures they are all validated in a single place.
   - This makes the code easier to read and maintain.

3. **Prepared for Reuse:**
   - The retrieved environment variables like `table_name_list_str`, `csv_output_prefix`, and `output_path` are available to the rest of the program for further use.

4. **Error Handling:**
   - All `env::var` calls are wrapped with proper error handling via `map_err`.

---

### Notes:
- If `TABLE_NAME` and `CSV_OUTPUT_PREFIX` are arrays/lists, consider parsing them accordingly (e.g., splitting values if the string is comma-separated).
- The `full_database_url` construction assumes several environment variables (`DB_PROTOCOL`, `DB_USER`, etc.) which are placeholders here and should be replaced with actual variable names relevant to your application.

This cleanup ensures your code is efficient, clear, and avoids unnecessary repetition.

**Review by rctz — COMMENTED** [review]
{
"body": "Critical Issue: Excessive redundancy for currect_dir.push and env::var calls, impacting readability and maintainability. Suggested Improvements: Consolidate these repeated blocks into reusable functions or constants for efficiency. Nitpick: Consider refactoring long error message strings into constants for reuse. Tests & Docs: Ensure that edge cases for missing environment variables are covered in tests.",
"event": "COMMENT",
"comments": [
{
"path": "src/main.rs",
"position": "20-21",
"body": "Redundant currect_dir.push('.env') calls detected, consolidate into one."
},
{
"path": "src/main.rs",
"position": "43-58",
"body": "Multiple env::var calls with identical keys are redundant; consider creating a function to fetch and validate environment variables."
},
{
"path": "src/main.rs",
"position": "43-58",
"body": "Repeated long error messages. Suggest placing these strings into reusable constants."
}
]
}

The provided code snippet includes significant redundancy in adding `.env` entries and repeated calls to retrieve environment variables with identical keys. This hampers readability and scalability. Consolidating these operations into reusable functions or constants is strongly recommended for clean, maintainable code.

**Review by rctz — COMMENTED** [review]
{
"body": "Critical Issue: Excessive redundancy for currect_dir.push and env::var calls, impacting readability and maintainability. Suggested Improvements: Consolidate these repeated blocks into reusable functions or constants for efficiency. Nitpick: Consider refactoring long error message strings into constants for reuse. Tests & Docs: Ensure that edge cases for missing environment variables are covered in tests.",
"event": "COMMENT",
"comments": [
{
"path": "src/main.rs",
"position": "20-21",
"body": "Redundant currect_dir.push('.env') calls detected, consolidate into one."
},
{
"path": "src/main.rs",
"position": "43-58",
"body": "Multiple env::var calls with identical keys are redundant; consider creating a function to fetch and validate environment variables."
},
{
"path": "src/main.rs",
"position": "43-58",
"body": "Repeated long error messages. Suggest placing these strings into reusable constants."
}
]
}

The provided code snippet includes significant redundancy in adding `.env` entries and repeated calls to retrieve environment variables with identical keys. This hampers readability and scalability. Consolidating these operations into reusable functions or constants is strongly recommended for clean, maintainable code.

**Review by rctz — COMMENTED** [review]
Hi, please check comments.

File: src/main.rs [hunk 1]
```diff
@@ -20,6 +20,8 @@ async fn main() -> Result<(), Box<dyn Error>> {
         .to_path_buf();
 
     currect_dir.push(".env");
+    currect_dir.push(".env");
+    currect_dir.push(".env");
```
Comment (rctz) [inline src/main.rs:25]: Multiple redundant .env pushes detected. Only one is necessary—please remove duplicates.

File: src/main.rs [hunk 2]
```diff
@@ -41,8 +43,31 @@ async fn main() -> Result<(), Box<dyn Error>> {
...
+    let csv_output_prefix = env::var("CSV_OUTPUT_PREFIX")
+        .map_err(|_| "Missing environment variable: CSV_OUTPUT_PREFIX")?;
     let output_path =
         env::var("OUTPUT_PATH").map_err(|_| "Missing environment variable: OUTPUT_PATH")?;
+        let csv_output_prefix = env::var("CSV_OUTPUT_PREFIX")
+        .map_err(|_| "Missing environment variable: CSV_OUTPUT_PREFIX")?;
+    let table_name_list_str =
+        env::var("TABLE_NAME").map_err(|_| "Missing environment variable: TABLE_NAME")?;
+    let csv_output_prefix = env::var("CSV_OUTPUT_PREFIX")
+        .map_err(|_| "Missing environment variable: CSV_OUTPUT_PREFIX")?;
```
Comment (rctz) [inline src/main.rs:69]: Repeated env::var calls found for TABLE_NAME and CSV_OUTPUT_PREFIX. Consolidate these calls into variables or a loop to minimize redundancy and improve clarity.

**Review by rctz — COMMENTED** [review]
Hi, please check comments.

File: src/main.rs (hunk 1 above)
Comment (rctz) [inline src/main.rs:25]: Multiple redundant .env pushes detected. Only one is necessary—please remove duplicates.

File: src/main.rs (hunk 2 above)
Comment (rctz) [inline src/main.rs:69]: Repeated env::var calls found for TABLE_NAME and CSV_OUTPUT_PREFIX. Consolidate these calls into variables or a loop to minimize redundancy and improve clarity.

**Review by rctz — COMMENTED** [review]
Hi, please check comments.

**Comment by rctz** [comment]
review
